File: app/src/main/java/com/hypnoweb/hypnowebapp/SplashScreenActivity.java
package com.hypnoweb.hypnowebapp;

import android.app.Activity;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.Nullable;

import com.hypnoweb.hypnowebapp.databinding.ActivityMainBinding;
import com.hypnoweb.hypnowebapp.di.HypnoWebApplication;
import com.hypnoweb.hypnowebapp.splashscreen.SplashViewModel;

import javax.inject.Inject;

public class SplashScreenActivity extends Activity {

    @Inject
    SplashViewModel splashViewModel;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ((HypnoWebApplication) getApplication()).getAppComponent().inject(this);
        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        binding.setViewModel(splashViewModel);
    }


    @Override
    protected void onStart() {
        super.onStart();
        splashViewModel.onStart();
    }

    @Override
    protected void onStop() {
        super.onStop();
        splashViewModel.onStop();
    }
}

File: app/src/main/java/com/hypnoweb/hypnowebapp/ui/splashscreen/SplashScreenActivity.kt
package com.hypnoweb.hypnowebapp.ui.splashscreen

import android.os.Bundle
import com.hypnoweb.hypnowebapp.R
import com.hypnoweb.hypnowebapp.ui.splashscreen.SplashScreenFragment
import dagger.android.support.DaggerAppCompatActivity

class SplashScreenActivity : DaggerAppCompatActivity() {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        this.supportFragmentManager.beginTransaction().replace(R.id.container, SplashScreenFragment.getInstance()).commit()
    }
}
